platform/ios/src/MGLPinAnnotationView.m

```diff
@@ -0,0 +1,96 @@
+#import "MGLPinAnnotationView.h"
+
+@implementation MGLPinAnnotationView
+
+- (instancetype)initWithReuseIdentifier:(nullable NSString *)reuseIdentifier {
+    self = [super initWithReuseIdentifier:reuseIdentifier];
+    _shadowColor = [UIColor blackColor];
+    _pinColor = [UIColor blueColor];
+    _strokeColor = [UIColor whiteColor];
+    _innerColor = [UIColor whiteColor];
+    self.backgroundColor = [UIColor clearColor];
+
+    return self;
+}
+
+- (void)drawRect:(CGRect)rect {
+    [self drawMarkerWithFrame:rect innerColor:_innerColor shadowColor:_shadowColor pinColor:_pinColor strokeColor:_strokeColor];
+}
+
+- (void)setShadowColor:(UIColor *)shadowColor {
+    _shadowColor = shadowColor;
+    [self setNeedsDisplay];
+}
+
+- (void)setPinColor:(UIColor *)pinColor {
+    _pinColor = pinColor;
+    [self setNeedsDisplay];
+}
+
+- (void)setStrokeColor:(UIColor *)strokeColor {
+    _strokeColor = strokeColor;
+    [self setNeedsDisplay];
+}
+
+- (void)setInnerColor:(UIColor *)innerColor {
+    _innerColor = innerColor;
+    [self setNeedsDisplay];
+}
+
+- (void)drawMarkerWithFrame:(CGRect)frame innerColor:(UIColor*)innerColor shadowColor:(UIColor*)shadowColor pinColor:(UIColor*)pinColor strokeColor:(UIColor*)strokeColor
+{
+    //// General Declarations
+    CGContextRef context = UIGraphicsGetCurrentContext();
+    
+    //// Group
+    {
+        //// Oval 2 Drawing
```

Comment:
Where possible, update these comments to match the specific parts of the marker that they represent.

Reply:
We should name these groups in PaintCode and perhaps generate a new file and leave it intact _or_ refactor the draw code to match our coding style.